### PR TITLE
Add Api endpoints to blog

### PR DESCRIPTION
**Pull request by alexiscyber14**
Requirements:
Create an API endpoint to list all posts for a user.
Create an API endpoint to list all comments for a user's post.
Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  module V1
+    class PostsController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_post, only: [:show]
+
+      def index
+        user = User.find(params[:user_id])
+        posts = user.posts
+
+        render json: posts
```
Comment (FrederickMih) [inline app/controllers/api/v1/posts_controller.rb:11]: - [ ] Hi @alexiscyber14 after successfully registering, and creating posts on your app, the API endpoint didn't return all posts in JSON format as can see in the screenshot below. I think I'm not supposed to register the user at his level. Please kindly fix :wrench: this. API endpoint should return the posts and comments in a JSON format. API endpoint should equally add a comment to a post.
Screenshot from 2023-06-29 09-39-36

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @alexiscyber14, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights :green_circle: 

- Great job for versioning your URL so as to be identified as an API.


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/posts_controller.rb (hunk 1 above)
Comment (alexiscyber14) [reply app/controllers/api/v1/posts_controller.rb:10]: @FrederickMih Done

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🎉🍾**
Hello @alexiscyber14 and @SamTush 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**